### PR TITLE
#671 rename with failure hint

### PR DESCRIPTION
**Pull request by stevenlmcgraw**
Draft pull request for #671 
______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: core/atrium-core-common/src/main/kotlin/ch/tutteli/atrium/assertions/builders/descriptiveWithFailureHint.kt [hunk 1]
```diff
@@ -27,20 +40,50 @@ fun Descriptive.DescriptionOption<Descriptive.FinalStep>.withFailureHint(
...
+ */
+//TODO if we introduce Record or something else as replacement for Assertion then not but if we keep Assertion
+// then move to logic and expect ProofContainer with 0.18.0
+@Deprecated("Use withHelpOnFailureBasedOnDefinedSubject; will be removed with 1.0.0")
 fun <T> Descriptive.DescriptionOption<Descriptive.FinalStep>.withFailureHintBasedOnDefinedSubject(
     expect: Expect<T>,
     failureHintFactory: (T) -> Assertion
 ): Descriptive.DescriptionOption<DescriptiveAssertionWithFailureHint.FinalStep> {
-    return withFailureHintBasedOnSubject(expect) {
+    return withHelpOnFailureBasedOnSubject(expect) {
```
Comment (robstoll) [inline core/atrium-core-common/src/main/kotlin/ch/tutteli/atrium/assertions/builders/descriptiveWithFailureHint.kt:86]: I would delegate to `withHelpOnFailureBasedOnDefinedSubject` but also fine like this

**Comment by robstoll** [comment]
@stevenlmcgraw looks good to me. Only a small thing which you can improve in case you do other things in addition (I did not have a look if you replaced everything). If you have done everything, then there is no need for the improvement and we can merge your first PR :)

File: core/atrium-core-common/src/main/kotlin/ch/tutteli/atrium/assertions/builders/descriptiveWithFailureHint.kt (hunk 1 above)
Comment (stevenlmcgraw) [reply core/atrium-core-common/src/main/kotlin/ch/tutteli/atrium/assertions/builders/descriptiveWithFailureHint.kt:86]: Hey @robstoll! I can change that if you would like for me to do so - I was just flipping any old `with*failureHint` method names to the new equivalent on all of the usages I could find to make sure I didn't introduce any unexpected changes. I'll adjust this on your signal.

**Comment by stevenlmcgraw** [comment]
Howdy @robstoll ! I used the same approach you suggested to another individual in the comments for this issue, i.e., I searched the whole project in IntelliJ with the `with.*Failure` regex. I _think_ I got them all, but I will sweep back through it later today to make sure I didn't miss anything.

**Comment by robstoll** [comment]
@stevenlmcgraw sounds good, let's wait to see if the build is green as well.

**Comment by robstoll** [comment]
@stevenlmcgraw don't worry about the coverage. It's perfect if you write a test for it but it's also OK if we keep it like that

**Comment by stevenlmcgraw** [comment]
@robstoll Gotcha - I don't think I would be able to get to working on some unit tests until later this week. I don't mind taking a crack at it, but I don't want to hold you up if you are looking to get this merged as quickly as possible.

**Comment by robstoll** [comment]
@stevenlmcgraw I am merging it already but a test would still be welcome. You can create a second PR with it.

In this sense, thanks for your first contribution to Atrium :+1:  and don't forget to :star: the repo if you liked it :slightly_smiling_face:

**Comment by stevenlmcgraw** [comment]
Thanks @robstoll ! I starred it. I'll try to set aside some time outside of work to do the tests. I'll be looking at some more issues here in Atrium as well - catch you on the next one!

**Comment by robstoll** [comment]
@stevenlmcgraw ping :wink: we could use your help with other issues. Let me know in case you need help

**Pull request merged**